SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorPackageAssembly.cpp

```diff
@@ -149,19 +151,43 @@ void FSpatialGDKPackageAssembly::OnTaskCompleted(int32 TaskResult)
 		{
 			FString NotificationMessage = FString::Printf(TEXT("Failed assembly upload to project: %s"), *FSpatialGDKServicesModule::GetProjectName());
 			ShowTaskEndedNotification(NotificationMessage, SNotificationItem::CS_Fail);
+			if (Status == EPackageAssemblyStatus::ASSEMBLY_EXISTS)
+			{
+				FMessageDialog::Open(EAppMsgType::Ok, LOCTEXT("Assembly_Exists", "The assembly has previously been uploaded enable the 'Force Overwrite on Upload' option in the Cloud Deployment dialog to overwrite the existing assembly."));
+			}
+			else if (Status == EPackageAssemblyStatus::BAD_PROJECT_NAME)
+			{
+				FMessageDialog::Open(EAppMsgType::Ok, LOCTEXT("Bad_Project_Name", "The project name appears to be incorrect or you do not have permissions for this project, you can edit the project name from the Cloud Deployment dialog."));
+			}
+			else if(Status == EPackageAssemblyStatus::NONE)
+			{
+				Status = EPackageAssemblyStatus::UNKOWN_ERROR;
+			}
 		});
 		Steps.Empty();
 	}
 }
 
 void FSpatialGDKPackageAssembly::OnTaskOutput(FString Message)
 {
+	//UNR-3486 parse for assembly name conflict so we can display a message to the user
+	//because the spatial cli doesn't return error codes this is done via string matching
+	if (Message.Find(TEXT("Either change the name or use the '--force' flag")) >= 0)
```

Comment:
I don't believe we pin against a specific spatial CLI version so this message could change. Perhaps we can get the team who owns it to include an error code we can rely on?

Reply:
yes that would be good, alternatively look for keywords like `--force` instead of the full sentence

Reply:
So, [ENG-3300](https://improbableio.atlassian.net/browse/ENG-3300) is meant to address the consistency concern. I would generally say the changing of error message wording is infrequent and we should be wary of putting too much effort on "future proofing" against a wording change. I prefer the bulk of the sentence, again we also test if the project name may be bad and show a similar message, narrowing this down to a small subset of keywords feels like overkill.

Reply:
I don't think searching for keywords is overkill. Instead of searching for a whole sentence, it would make more sense imo to simply look for 1-2 words that clearly indicate which error message it is supposed to be. This is less error prone in case the message changes.

Reply:
So overall I believe the solution is weak, parsing strings to glean error types is poor design. I don't know whether ENG-3300 will be done, I hope that at least we will influence the design of whatever tool/API IMP will use to integrate with UnrealGDK so that it is designed with a clean error channel. And a commitment that the enumerated errors will remain consistent (though of course I'd expect new ones to be added over time). Searching for "keywords" in my interpretation means searching for multiple different values, which involves more complicated code to iterate the search list and validate each specific word is found, which seems far more trouble than it's worth.